llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -6141,6 +6138,68 @@ static SDValue BuildExactSDIV(const TargetLowering &TLI, SDNode *N,
   return DAG.getNode(ISD::MUL, dl, VT, Res, Factor);
 }
 
+/// Given an exact UDIV by a constant, create a multiplication
+/// with the multiplicative inverse of the constant.
```

Comment:
Is there a reference for what this is doing somewhere?

Reply:
> Is there a reference for what this is doing somewhere?



> Is there a reference for what this is doing somewhere?

yes. hacker's delight, but this matches exactly what SDiv does too, since the algorithm is the same.